### PR TITLE
Reworked docs for config options

### LINKED ISSUE #57: Document transform parameters
See https://github.com/marklogic-community/kafka-marklogic-connector/issues/56#issuecomment-1201853124

### PR DESCRIPTION
**Pull request by rjrudin**
Resolves #57. 

- Updated docs for each config option so they're the same in all 3 places - README, marklogic-sink.properties, and MarkLogicSinkConfig
- Updated the CONTRIBUTING guide for testing with Apache Kafka
- Updated the README so it is focused on using the connectors as opposed to developing/testing them (more updates will happen to this before 1.7.0 is released)

### CODE REVIEW

File: AWS-CloudFormation/s3Resources/marklogic-sink.properties [hunk 1]
```diff
@@ -2,105 +2,21 @@
...
-
-# Should only need one task since it's using a WriteBatcher, which is multi-threaded
 tasks.max=1
-
 # Topics to consume from [comma separated list for multiple topics]
 topics=marklogic
 
-
 # MarkLogic connector-specific properties
+# See ./config/marklogic-sink.properties for information on each of these
```
Comment (rjrudin) [inline AWS-CloudFormation/s3Resources/marklogic-sink.properties:10]: Simplified this file so we don't need to keep the config docs in sync with the main marklogic-sink.properties file

File: config/marklogic-sink.properties [hunk 2]
```diff
@@ -1,113 +1,130 @@
...
-ml.connection.type=
+# MarkLogic username for 'DIGEST' and 'BASIC' authentication
+ml.connection.username=
 
-# Either DIGEST, BASIC, CERTIFICATE, KERBEROS, or NONE
-ml.connection.securityContextType=DIGEST
+# MarkLogic password for 'DIGEST' and 'BASIC' authentication
+ml.connection.password=
+
+# Path to PKCS12 file for 'CERTIFICATE' authentication
```
Comment (rjrudin) [inline config/marklogic-sink.properties:34]: Commenting out anything not used in this file. The README should be the authoritative source on properties and default values. This file instead has some additional values set to simplify testing the connector.

File: src/main/java/com/marklogic/kafka/connect/DefaultDatabaseClientConfigBuilder.java [hunk 3]
```diff
@@ -47,13 +47,13 @@ public DatabaseClientConfig buildDatabaseClientConfig(Map<String, Object> parsed
             clientConfig.setPassword(password.value());
         }
         clientConfig.setPort((Integer) parsedConfig.get(MarkLogicSinkConfig.CONNECTION_PORT));
-        Boolean customSsl = (Boolean) parsedConfig.get(MarkLogicSinkConfig.SSL);
+        Boolean customSsl = (Boolean) parsedConfig.get(MarkLogicSinkConfig.ENABLE_CUSTOM_SSL);
```
Comment (rjrudin) [inline src/main/java/com/marklogic/kafka/connect/DefaultDatabaseClientConfigBuilder.java:50]: Renamed this as "SSL" was a bit vague.

File: src/main/java/com/marklogic/kafka/connect/DefaultDatabaseClientConfigBuilder.java [hunk 4]
```diff
@@ -133,13 +126,8 @@ protected DatabaseClientConfig configureCustomSslConnection(DatabaseClientConfig
                 }
                 KeyManager[] key = keyManagerFactory.getKeyManagers();
                 try {
-                    if (tlsVersion != null && tlsVersion.trim().length() > 0) {
-                        sslContext = SSLContext.getInstance(tlsVersion);
-                    } else {
-                        sslContext = SSLContext.getInstance("TLSv1.2");
-                    }
+                    sslContext = SSLContext.getInstance(tlsVersion);
```
Comment (rjrudin) [inline src/main/java/com/marklogic/kafka/connect/DefaultDatabaseClientConfigBuilder.java:129]: Simplified this logic, as we have a default value of "TLSv1.2" in place already.

**Review by BillFarber — APPROVED** [review]

**Review by anu3990 — APPROVED** [review]

**Pull request merged**